mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -103,18 +103,27 @@ pub const WATCHER_MESSAGE_SENT_LOG: &str = "Watcher message sent...";
 pub const MAKER_PAYMENT_SPENT_BY_WATCHER_LOG: &str = "Maker payment is spent by the watcher...";
 
 #[cfg(not(target_arch = "wasm32"))]
-pub fn stats_taker_swap_dir(ctx: &MmArc) -> PathBuf { ctx.dbdir().join("SWAPS").join("STATS").join("TAKER") }
+pub fn stats_taker_swap_dir(ctx: &MmArc) -> PathBuf {
+    #[cfg(not(feature = "new-db-arch"))]
+    return ctx.dbdir().join("SWAPS").join("STATS").join("TAKER");
+    #[cfg(feature = "new-db-arch")]
+    return ctx.global_dir().join("SWAPS").join("STATS").join("TAKER");
+}
 
 #[cfg(not(target_arch = "wasm32"))]
 pub fn stats_taker_swap_file_path(ctx: &MmArc, uuid: &Uuid) -> PathBuf {
     stats_taker_swap_dir(ctx).join(format!("{}.json", uuid))
 }
 
 async fn save_my_taker_swap_event(ctx: &MmArc, swap: &TakerSwap, event: TakerSavedEvent) -> Result<(), String> {
-    let swap = match SavedSwap::load_my_swap_from_db(ctx, swap.uuid).await {
+    let maker_coin_pub = swap.my_maker_coin_htlc_pub();
+    let maker_coin_address = try_s!(swap.maker_coin.address_from_pubkey(&maker_coin_pub));
+    let swap = match SavedSwap::load_my_swap_from_db(ctx, Some(&maker_coin_address), swap.uuid).await {
         Ok(Some(swap)) => swap,
         Ok(None) => SavedSwap::Taker(TakerSavedSwap {
             uuid: swap.uuid,
+            #[cfg(all(not(target_arch = "wasm32"), feature = "new-db-arch"))]
+            address_dir: maker_coin_address,
```

Comment:
IMO using the name `address_dir` in `TakerSavedSwap` doesn't make too much, we can just call it `address` to keep it general-purpose.

```suggestion
    let maker_coin_address = try_s!(swap.maker_coin.address_from_pubkey(&maker_coin_pub));
    let swap = match SavedSwap::load_my_swap_from_db(ctx, Some(&maker_coin_address), swap.uuid).await {
        Ok(Some(swap)) => swap,
        Ok(None) => SavedSwap::Taker(TakerSavedSwap {
            uuid: swap.uuid,
            #[cfg(all(not(target_arch = "wasm32"), feature = "new-db-arch"))]
            address: maker_coin_address,
```

Reply:
address alone is ambiguous (since each swap party has two addresses). what about `maker_address`?

Reply:
That can work too.

Reply:
[b896af6](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2398/commits/b896af6bcf6a3632e9734ca7f103ba8da43c7c81)